File: DobotDemoV2.0/Depracate/DobotDemoForVB/DobotDemoForVB/VBDll/DobotDllType.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Runtime.InteropServices
Imports System.Text

Namespace DobotDemoForVB
    Public Enum EndType
        EndTypeCustom
        EndTypeSuctionCap
        EndTypeGripper
        EndTypeLaser
        EndTypePen
        EndTypeMax
        EndTypeNum = EndTypeMax
    End Enum

    <StructLayout(Runtime.InteropServices.LayoutKind.Sequential, Pack:=1)> _
    Public Structure EndEffectorParams
        Dim xBias As Single
        Dim yBias As Single
        Dim zBias As Single
    End Structure

    Public Structure Pose
        Dim x As Single
        Dim y As Single
        Dim z As Single
        Dim r As Single
        Dim joint1Angle As Single
        Dim joint2Angle As Single
        Dim joint3Angle As Single
        Dim joint4Angle As Single
    End Structure

    Public Structure HOMEParams
        Dim x As Single
        Dim y As Single
        Dim z As Single
        Dim r As Single
    End Structure

    Public Structure HOMECmd
        Dim temp As UInt32
    End Structure

    Public Structure JOGJointParams
        Dim velocity1 As Single
        Dim velocity2 As Single
        Dim velocity3 As Single
        Dim velocity4 As Single

        Dim acceleration1 As Single
        Dim acceleration2 As Single
        Dim acceleration3 As Single
        Dim acceleration4 As Single
    End Structure

    Public Structure JOGCoordinateParams
        Dim velocity1 As Single
        Dim velocity2 As Single
        Dim velocity3 As Single
        Dim velocity4 As Single

        Dim acceleration1 As Single
        Dim acceleration2 As Single
        Dim acceleration3 As Single
        Dim acceleration4 As Single
    End Structure

    Public Structure JOGCommonParams
        Dim velocityRatio As Single
        Dim accelerationRatio As Single
    End Structure

    Public Enum JogCmdType
        JogIdle
        JogAPPressed
        JogANPressed
        JogBPPressed
        JogBNPressed
        JogCPPressed
        JogCNPressed
        JogDPPressed
        JogDNPressed
        JogEPPressed
        JogENPressed
    End Enum


    <StructLayout(Runtime.InteropServices.LayoutKind.Sequential, Pack:=1)> _
    Public Structure JogCmd
        Dim isJoint As Byte
        Dim cmd As Byte
    End Structure

    Public Structure PTPJointParams
        Dim velocity1 As Single
        Dim velocity2 As Single
        Dim velocity3 As Single
        Dim velocity4 As Single

        Dim acceleration1 As Single
        Dim acceleration2 As Single
        Dim acceleration3 As Single
        Dim acceleration4 As Single
    End Structure

    Public Structure PTPCoordinateParams
        Dim xyzVelocity As Single
        Dim rVelocity As Single
        Dim xyzAcceleration As Single
        Dim rAcceleration As Single
    End Structure

    Public Structure PTPJumpParams
        Dim jumpHeight As Single
        Dim zLimit As Single
    End Structure

    Public Structure PTPCommonParams
        Dim velocityRatio As Single
        Dim accelerationRatio As Single
    End Structure

    ' For play back
    Public Enum PTPMode
        PTPJUMPXYZMode
        PTPMOVJXYZMode
        PTPMOVLXYZMode

        PTPJUMPANGLEMode
        PTPMOVJANGLEMode
        PTPMOVLANGLEMode

        PTPMOVJXYZINCMode
        PTPMOVLXYZINCMode
    End Enum

    <StructLayout(Runtime.InteropServices.LayoutKind.Sequential, Pack:=1)> _
    Public Structure PTPCmd
        Dim ptpMode As Byte
        Dim x As Single
        Dim y As Single
        Dim z As Single
        Dim r As Single
    End Structure

    Public Structure CPParams
        Dim planAcc As Single
        Dim juncitionVel As Single
        Dim acc As Single
        Dim realTimeTrack As Byte
    End Structure

    Public Enum ContinuousPathMode
        CPRelativeMode
        CPAbsoluteMode
    End Enum

    <StructLayout(Runtime.InteropServices.LayoutKind.Sequential, Pack:=1)> _
    Public Structure CPCmd
        Dim cpMode As Byte
        Dim x As Single
        Dim y As Single
        Dim z As Single
        Dim velocity As Single
    End Structure

    Public Structure WAITCmd
        Dim timeout As UInt32
    End Structure

    Public Enum IOFunction
        IOFunctionDummy
        IOFunctionDO
        IOFunctionPWM
        IOFunctionDI
        IOFunctionADC
    End Enum

    ' API result
    Public Enum DobotConnect
        DobotConnect_NoError
        DobotConnect_NotFound
        DobotConnect_Occupied
    End Enum

    Public Enum DobotCommunicate
        DobotCommunicate_NoError
        DobotCommunicate_BufferFull
        DobotCommunicate_Timeout
    End Enum
End Namespace

File: DobotDemoV2.0/Depracate/DobotDemoForC#/DobotClientDemo2.0/CPlusDll/DobotDllType.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Runtime.InteropServices;
using System.Text;

namespace DobotClientDemo.CPlusDll
{
    
    /*
     * 末端选择
     */
    public enum EndType {
        EndTypeCustom,
        EndTypeSuctionCap,
        EndTypeGripper,
        EndTypeLaser,
        EndTypePen,
        EndTypeMax,
        EndTypeNum = EndTypeMax
    };

    [StructLayout(LayoutKind.Sequential, CharSet=CharSet.Ansi, Pack=1)]
    public struct EndTypeParams {
         public float xBias;
         public float yBias;
         public float zBias;
    };

    /*
     * 位姿
     */
    public struct Pose {
        public float x;
        public float y;
        public float z;
        public float rHead;
        [MarshalAs(UnmanagedType.ByValArray, SizeConst = 4)]
        public float[] jointAngle;
    };

    /*
     * 运动学参数
     */
    public struct Kinematics {
        public float velocity;
        public float acceleration;
    };

    /*
     * 报警状态：暂时支持32种报警状态
     */
    public struct AlarmsState {
        public int alarmsState;
    };

    /*********************************************************************************************************
    ** HOME参数
    *********************************************************************************************************/
    public struct HOMEParams {
        public float x;
        public float y;
        public float z;
        public float r;
    };

    public struct HOMECmd {
        public int temp;
    };

    /*********************************************************************************************************
    ** 点动示教部分
    *********************************************************************************************************/
    /*
     * 单关节点动示教参数
     */
    public struct JOGJointParams
    {
        [MarshalAs(UnmanagedType.ByValArray, SizeConst = 4)]
        public float[] velocity;
        [MarshalAs(UnmanagedType.ByValArray, SizeConst = 4)]
        public float[] acceleration;
    };

    /*
     * 单坐标轴点动示教参数
     */
    public struct JOGCoordinateParams
    {
        [MarshalAs(UnmanagedType.ByValArray, SizeConst = 4)]
        public float[] velocity;
        [MarshalAs(UnmanagedType.ByValArray, SizeConst = 4)]
        public float[] acceleration;
    };

    /*
     * 点动示教公共参数
     */
    public struct JOGCommonParams
    {
        public float velocityRatio;
        public float accelerationRatio;
    };

    /*
     * Jog Cmd
     */
    public enum JogCmdType{
        JogIdle,
        JogAPPressed,
        JogANPressed,
        JogBPPressed,
        JogBNPressed,
        JogCPPressed,
        JogCNPressed,
        JogDPPressed,
        JogDNPressed,
        JogEPPressed,
        JogENPressed
    };

    /*
     * Jog instant cmd
     */
    [StructLayout(LayoutKind.Sequential, CharSet = CharSet.Ansi, Pack = 1)]
    public struct JogCmd {
        public byte isJoint;
        public byte cmd;
    };

    /*********************************************************************************************************
    ** 再现运动部分
    *********************************************************************************************************/
    /*
     * 再现运动参数
     */
    public struct PTPJointParams
    {
        [MarshalAs(UnmanagedType.ByValArray, SizeConst = 4)]
        public float[] velocity;
        [MarshalAs(UnmanagedType.ByValArray, SizeConst = 4)]
        public float[] acceleration;
    };
    public struct PTPCoordinateParams
    {
        public float xyzVelocity;
        public float rVelocity;
        public float xyzAcceleration;
        public float rAcceleration;
    };

    public struct PTPJumpParams
    {
        public float jumpHeight;
        public float zLimit;
    };

    public struct PTPCommonParams
    {
        public float velocityRatio;
        public float accelerationRatio;
    };

    // For play back
    public enum PTPMode {
        PTPJUMPXYZMode,
        PTPMOVJXYZMode,
        PTPMOVLXYZMode,

        PTPJUMPANGLEMode,
        PTPMOVJANGLEMode,
        PTPMOVLANGLEMode,

        PTPMOVJXYZINCMode,
        PTPMOVLXYZINCMode,
    };

    [StructLayout(LayoutKind.Sequential, CharSet = CharSet.Ansi, Pack = 1)]
    public struct PTPCmd
    {
        public byte ptpMode;
        public float x;
        public float y;
        public float z;
        public float rHead;
    };

    /*********************************************************************************************************
    ** 连续轨迹：Continuous path
    *********************************************************************************************************/
    /*
     * CP参数
     */
    public struct CPParams
    {
        public float planAcc;
        public float juncitionVel;
        public float acc;
        public byte realTimeTrack;
    };

    public enum ContinuousPathMode {
        CPRelativeMode,
        CPAbsoluteMode
    };
    [StructLayout(LayoutKind.Sequential, CharSet = CharSet.Ansi, Pack = 1)]
    public struct CPCmd
    {
        public byte cpMode;
        public float x;
        public float y;
        public float z;
        public float velocity;
    };

    /*********************************************************************************************************
    ** User parameters
    *********************************************************************************************************/

    public struct WAITCmd
    {
        public UInt32 timeout;
    };

    public enum IOFunction
    {
        IOFunctionDummy,
        IOFunctionDO,
        IOFunctionPWM,
        IOFunctionDI,
        IOFunctionADC
    };

    public struct  IOMultiplexing{
        public byte address;
        public byte multiplex;
    };

    public struct IODO{
        public byte  address;
        public byte  level;
    };

    public struct IOPWM {
        public byte address;
        public float frequency;
        public float dutyCycle;
    };

    public struct IODI {
        public byte address;
        public byte level;
    };

    public struct IOADC
    {
        public byte address;
        public UInt16 value;
    };

    /*
     * ARC related
     */
    public struct ARCParams
    {
        public float xyzVelocity;
        public float rVelocity;
        public float xyzAcceleration;
        public float rAcceleration;
    };

    public struct ARCCmd {
        public float cirPoint_x;
        public float cirPoint_y;
        public float cirPoint_z;
        public float cirPoint_r;

        public float toPoint_x;
        public float toPoint_y;
        public float toPoint_z;
        public float toPoint_r;
    };

    /*********************************************************************************************************
    ** User parameters
    *********************************************************************************************************/
    public struct UserParams {
        [MarshalAs(UnmanagedType.ByValArray, SizeConst = 8)]
        public float[] param;
    };

    /*********************************************************************************************************
    ** API result
    *********************************************************************************************************/
    public enum DobotConnect{
        DobotConnect_NoError,
        DobotConnect_NotFound,
        DobotConnect_Occupied
    };

    public enum DobotCommunicate{
        DobotCommunicate_NoError,
        DobotCommunicate_BufferFull,
        DobotCommunicate_Timeout,
        DobotCommunicate_InvalidParams
    };
}
